examples/samples_test.go

```diff
@@ -0,0 +1,41 @@
+package examples
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+	"time"
+)
+
+func TestRunSamples(t *testing.T) {
+	items, _ := os.ReadDir(".")
+	for _, item := range items {
+		if item.IsDir() {
+			fmt.Printf("Running sample %v\n", item.Name())
+			mainFile, err := os.Stat(path.Join(item.Name(), "main.go"))
+			if err == nil && !mainFile.IsDir() {
+				// Verify that we can run the sample.
+				if err := os.Chdir(item.Name()); err != nil {
+					t.Fatalf("failed to change current directory to %v: %v", item.Name(), err)
+				}
+				ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+				cmd := exec.CommandContext(ctx, "go", "run", "main.go")
```

Comment:
Here use:

```
cmd := exec.CommandContext(ctx, "go", "run", "-race", ".")
```

`-race` to find race issues. And `.` so that the package can use more than one file, if necessary.